LabApiUtilities/build.gradle

```diff
@@ -0,0 +1,70 @@
+plugins {
+    id 'java-library'
+    id 'com.microsoft.identity.buildsystem' version '0.1.0'
+    id 'com.github.gmazzo.buildconfig' version '3.0.0'
+}
+
+def labClientSecret = "" // will be blank unless specified by developer
+
+if (project.hasProperty("labSecret")) {
+    labClientSecret = labSecret
+}
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_1_7
+    targetCompatibility = JavaVersion.VERSION_1_7
+}
+
+buildSystem {
+    desugar = false
+}
+
+buildConfig {
+    packageName("com.microsoft.identity.labapi.utilities")  // forces the package. Defaults to '${project.group}'
+    useJavaOutput() // forces the outputType to 'java'
+}
+
+
+sourceSets {
+    main {
+        java.srcDirs = ['src/main']
+    }
+    test {
+        java.srcDirs = ['src/test', 'build/generated/source/buildConfig/test']
+        buildConfigField("String", "LAB_CLIENT_SECRET", "\"$labClientSecret\"")
+    }
+}
+
+dependencies {
+    // keyvault per product flavor
+    if (findProject(":keyvault") != null) {
+        //Use project reference if available
+        implementation(project(":keyvault")) {
+            transitive = true
+        }
+    } else {
+        //Always use the latest release
+        implementation(group: 'com.microsoft.identity', name: 'keyvault', version: '0.0.+')
+    }
+
+    // labapi per product flavor
+    if (findProject(":labapi") != null) {
+        //Use project reference if available
+        implementation(project(":labapi")) {
+            transitive = true
+        }
+    } else {
+        //Always use the latest release
+        implementation(group: 'com.microsoft.identity', name: 'labapi', version: '0.0.+')
+    }
+
+    implementation "com.nimbusds:nimbus-jose-jwt:$rootProject.ext.nimbusVersion"
+    implementation "com.microsoft.azure:msal4j:1.10.0"
```

Comment:
Looks like some versions are hardcoded and other reference potentially a versions file.  I'd consider adoption one or the other.

Reply:
Referencing from versions file now: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1389/commits/289f912db7cf5daacfb5ce79b5bd1b80441124bb